### DISCUSSION TITLE
The data type for the headers in the scope is represented as list in the docs other than tuples is it something worth updating?

### DISCUSSION THREAD

**Discussion by kaypee90** [Kludex/uvicorn#2072] [Q&A] [answered]
I realised the headers in the scope is returned as a list of tuples `'headers': [(b'content-type', b'text/plain'), (b'user-agent', b'PostmanRuntime/7.32.3'), (b'accept', b'*/*'), (b'accept-encoding', b'gzip, deflate, br'), (b'connection', b'keep-alive'), (b'content-length', b'54')]`

But the docs represent them as a list of lists:
{
    'type': 'http',
    'scheme': 'http',
    'root_path': '',
    'server': ('127.0.0.1', 8000),
    'http_version': '1.1',
    'method': 'GET',
    'path': '/',
    'headers': [
        [b'host', b'127.0.0.1:8000'],
        [b'user-agent', b'curl/7.51.0'],
        [b'accept', b'*/*']
    ]
}

Is it something that is worth updating?

**Comment by Kludex** [comment, marked as answer]
Both are compliant with the ASGI specification: https://asgi.readthedocs.io/en/latest/specs/www.html#http

If you want to open a PR to match what we have in our code, I will accept it... But again, nothing wrong here.

**Reply by kaypee90** [reply]
@Kludex Here is a PR to that effect https://github.com/encode/uvicorn/pull/2073
